src/routes/fbauth.js

```diff
@@ -0,0 +1,66 @@
+require('dotenv').config();
+const fbRouter = require('express').Router();
+const Admin = require('../models/admin');
+const User = require('../models/user');
+const passport = require('passport');
+const FacebookStrategy = require('passport-facebook').Strategy;
+const findOrCreate = require('mongoose-findorcreate');
+
+
+
+
+// use static serialize and deserialize of model for passport session support
+passport.serializeUser(function(user, done) {
+  done(null, user.id);
+});
+
+passport.deserializeUser(function(id, done) {
+
+  if(Admin){
+    Admin.findById(id, function(err, user) {
+      done(err, user);
+    });
+  }else{
+    User.findById(id, function(err, user) {
+      done(err, user);
+    });
+  }
+
```

Comment:
Boss, nice job here. But could you help explain what this part of the code doing per say especially the `callback` passed to the `passport.serializeUser` and `passport.deserializeUser`

Thanks.

Reply:
It's a way to add and remove user sessions on the server, you can add additional properties to that session from the user data gotten.

Reply:
ceoFred has answered serialization and deserialization is the addition and removal of cookies to a user for sessions.. the callback (user,done)=> user is the authenticated user to be serialized while done is to check if it's completed where it returns null if there was an error and user.id if it was successfull